### PR TITLE
Dicitonaries!!! :D

### PR DESCRIPTION
**Pull request by kirbyfan64**
Sorry this took so long!

Basically, this is how I ended up making it work:

- `2'at` creates an entry `{a: 2}`. Note that I had to reverse the key and value; otherwise, it would be impossible to tell this apart from the next form.
- `[2'a]t` creates an array of entries `[{a: 2}]`.
- `[2'a]t 'a !&` looks up `a` in the dictionary, raising an error if the lookup fails.
- `2 'a []t=` results in the dictionary `[{a: 1}]`. Again, note that I had to put the value and key *before* the array, otherwise `=` would fail to know whether or not to do a normal comparison.
- You can also use `=` with an already-created entry: `2 'a t []t=` works the same way as above.
- `(` gets the key from an entry, and `)` gets the value.

@phase Please review!

### CODE REVIEW

File: o.c [hunk 1]
```diff
@@ -65,15 +65,18 @@ V rev(ST s){P t;L i;for(i=0;i<s->p/2;++i){t=s->st[i];s->st[i]=s->st[s->p-i-1];s-
 ST rst=0; //root stack
 
 //objects
-typedef enum{TD,TS,TA,TCB}OT; //decimal,string,array,codeblock
-typedef struct{OT t;union{F d;struct{S s;L z;}s;ST a;};}OB;typedef OB*O; //type:type flag,value{decimal,{string,len},array}(NOTE:code blocks use string struct to store their code!)
+typedef enum{TD,TS,TA,TCB,TR}OT; //decimal,string,array,codeblock,entry
```
Comment (phase) [inline o.c]: Is `TE` taken? Why `TR`?

**Comment by phase** [comment]
> `2 'a []t=` results in the dictionary `[{a: 1}]`

This confuses me. If I were to put it into Lisp notation...

```lisp
(=            ; I have no clue what's going on here, nor why it produces 1
  (t [] 'a 2) ; I don't see what this is doing
)
```

File: o.c [hunk 2]
```diff
@@ -266,6 +278,13 @@ V uv(ST s,O o){if(o->t==TCB)excb(o);else psh(s,dup(o));} //execute the object if
 
 V bcv(ST s){ST r;I i=0,a,b;O ao,bo=pop(s);ao=pop(s);if(ao->t!=TD||bo->t!=TD)TE;a=ao->d;b=bo->d/*truncate*/;dlo(ao);dlo(bo);r=newst(BZ);while(a){C c=a%b+'0';if(c>'9')c+=7;psh(r,newod(a%b));if(b==1)--a;else a/=b;}psh(s,newoa(r));} //base conversion
 
+V entry(ST s){O r,v,k=pop(s);
```
Comment (phase) [inline o.c]: Is this method inserting entries into a dictionary?

**Comment by kirbyfan64** [comment]
> This confuses me.

Haha, typo. I meant `[{a: 1}]`. `=` inserts entries, so all that code is doing is inserting the given key and value into the array.

File: o.c (hunk 1 above)
Comment (kirbyfan64) [inline o.c]: `TE` is for type errors. `TR` is like ent**R**y, because `TT` looked worse...

**Comment by phase** [comment]
> > > `2 'a []t=` results in the dictionary **`[{a: 1}]`**

> > This confuses me.

> Haha, typo. I meant **`[{a: 1}]`**

:confused:?

**Comment by kirbyfan64** [comment]
@phase Ugh, mobile phone keyboards, I swear...

I meant `[{a: 2}]`. How did I mess that up so many times...

**Comment by phase** [comment]
So `[]t` creates an empty dictionary, and `=` inserts an entry of `{'a : 2}`?

I think `2 'a t [] =` makes more sense, as `t` creates entries?

**Comment by kirbyfan64** [comment]
Actually, `t` takes an array of elements and turns it into an array of entries. It functions like we had discussed in the main dictionary issue. In this case, it wasn't actually necessary; I don't know why exactly I put it there...

Actually, I just suck at explanations. It's basically like `c` from #56.

**Comment by phase** [comment]
:+1:

**Pull request merged**